lib/moped/protocol/reply.rb

```diff
@@ -108,7 +108,12 @@ def query_failed?
       # @since 1.2.10
       def unauthorized?
         result = documents[0]
-        result["code"] == UNAUTHORIZED || result["assertionCode"] == UNAUTHORIZED
+        return false if result.nil?
+        
+        err = error_message(result)
+        UNAUTHORIZED.include?(result["code"]) ||
```

Comment:
on one hand, I agree. On the other, I c/p'd from
https://github.com/mongoid/moped/commit/0416e5d9db552ebef299ed4748e7541cf38710fa
and kinda want to keep it the same. I dunno, thoughts?